docs/data/joy/components/autocomplete/InfiniteScrolling.js

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import Autocomplete from '@mui/joy/Autocomplete';
+
+export default function InfiniteScrolling() {
+  const [films, setFilms] = React.useState(allFilms.slice(0, 10));
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const handleOnScrollToBottom = () => {
+    if (allFilms.length !== films.length) {
+      setIsLoading(true);
+      setTimeout(() => {
+        setIsLoading(false);
+        setFilms((previousFilms) => [
+          ...previousFilms,
+          ...allFilms.slice(previousFilms.length, previousFilms.length + 10),
+        ]);
+      }, 2000);
+    }
+  };
+
+  return (
+    <Autocomplete
+      id="infinite-scrolling-demo"
+      options={films}
+      sx={{ width: 300 }}
+      loading={isLoading}
+      onScrollToBottom={handleOnScrollToBottom}
+    />
+  );
+}
+
+// Top 100 films as rated by IMDb users. http://www.imdb.com/chart/top
+const allFilms = [
```

Comment:
Instead of hard-coding the options, is it possible to generate random items when you scroll to the bottom? This way I don't have to reload the page to try it again.